lib/src/native/realm_core.dart

```diff
@@ -1811,59 +1870,64 @@ class ConfigHandle extends HandleBase<realm_config> {
 }
 
 class RealmHandle extends HandleBase<shared_realm> {
+  int _counter = 0;
+
+  final Map<int, WeakReference<RootedHandleBase>> _children = {};
```

Comment:
Could we use an `Expando` here instead?

Reply:
I don't see how - we need to be able to iterate over all keys of the collection and `Expando` doesn't expose an API to do that.